<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2025-02-04" />

<title>Reproducible scripting, AlphaSimR, Homework</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PLBRG_7170</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Reproducible scripting, AlphaSimR,
Homework</h1>
<h4 class="author">Jean-Luc Jannink</h4>
<address class="author_afil">
USDA-ARS / Cornell<br><h4 class="date">February 4, 2025</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-02-11
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>PLBRG_7170/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250128code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250128)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250128code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250128)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong5d117d5">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 5d117d5
</a>
</p>
</div>
<div id="strongRepositoryversionstrong5d117d5"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 5d117d5.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store

Untracked files:
    Untracked:  code/addToREADME.R
    Untracked:  output/QTLgenotypes.csv

Unstaged changes:
    Modified:   analysis/Lab3_ProgrammingAlphaSimR.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/Lab3_ProgrammingAlphaSimR.Rmd</code>) and HTML
(<code>docs/Lab3_ProgrammingAlphaSimR.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
5d117d5
</td>
<td>
tmulderrig
</td>
<td>
2025-02-04
</td>
<td>
lab 3
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<style>
pre {
  overflow-x: auto;
}
pre code {
  word-wrap: normal;
  white-space: pre;
}
</style>
<div id="preliminaries" class="section level1">
<h1>Preliminaries</h1>
<div id="rmarkdown" class="section level3">
<h3>Rmarkdown</h3>
<p>This lab is written in Rmarkdown. I find it challenging to remember
the syntax, but here is a <a
href="https://rstudio.github.io/cheatsheets/html/rmarkdown.html">cheatsheet</a></p>
</div>
<div id="learning-objectives" class="section level3">
<h3>Learning objectives</h3>
<ol style="list-style-type: decimal">
<li>A bit more on reproducible scripting<br />
Some other things that matter to me to ensure communicability and
reproducibility</li>
</ol>
<ol style="list-style-type: lower-roman">
<li>The main README.md file should contain information about the
conditions under which the script was run<br />
</li>
<li>I like the script to have a predictable strucure: load and document
the packages used, if there is a stochastic component, set the random
seed, set and document the parameter values that will affect the script
output<br />
</li>
<li>Use a consistent directory structure. Happily, such a consistent
structure is enforced by <code>workflowr</code><br />
</li>
<li>Use the package <code>here</code> to make file locations relative to
the base project folder</li>
</ol>
<ol start="2" style="list-style-type: decimal">
<li><p>AlphaSimR<br />
<code>AlphaSimR</code> is a package to simulate breeding populations and
tasks. It is not completely intuitive nor completely well-documented. We
will want to use it ultimately to optimize breeding schemes. As a way to
introduce AlphaSimR, this Rmarkdown script contains code to illustrate
the “outbreak of variation” that occurs when a heterozygous individual
is self-fertilized. This phenomenon was important in supporting the
notion that quantitative traits were influenced by many loci. Here is an
<a
href="https://cran.r-project.org/web/packages/AlphaSimR/vignettes/intro.html">introduction</a>
showing code to simulate a very simple mass selection program. Here is a
<a
href="https://cran.r-project.org/web/packages/AlphaSimR/vignettes/traits.pdf">deep
dive</a> into how <code>AlphaSimR</code> simulates traits.</p></li>
<li><p>Prepare a Homework<br />
Your homework will be to write a script that illustrates the interesting
phenomenon of regression toward the mean between parents and progeny
that we discussed in class.</p></li>
</ol>
</div>
</div>
<div id="ordering-of-the-script" class="section level1">
<h1>Ordering of the script</h1>
<p>It’s good to have all scripts in the same order with respect to
standard tasks</p>
<div id="load-packages-first" class="section level3">
<h3>Load packages first</h3>
<p>If your script depends on external packages, load them at the
beginning. This shows users early on what the script dependencies
are.</p>
<pre class="r"><code>packages_used &lt;- c(&quot;AlphaSimR&quot;, &quot;tidyverse&quot;, &quot;workflowr&quot;, &quot;here&quot;)
ip &lt;- installed.packages()
all_packages_installed &lt;- TRUE
for (package in packages_used){
  if (!(package %in% ip[,&quot;Package&quot;])){
    print(paste(&quot;Please install package&quot;, package))
    all_packages_installed &lt;- FALSE
  }
}#END packages_used
if (!all_packages_installed) stop(&quot;Need to install packages&quot;)

library(tidyverse)</code></pre>
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.4     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<p>Notice the conflicts report from loading tidyverse. Two packages
(<code>dplyr</code> and <code>stats</code>) both have a function called
<code>filter</code>. Since dplyr was loaded <em>after</em> stats, if you
use the function <code>filter</code>, it will go to the dplyr version.
It is dangerous to rely on what order packages have been loaded to
determine which <code>filter</code> function you get. R syntax to
prevent ambiguity is to write either <code>dplyr::filter</code> or
<code>stats::filter</code>. Using that syntax will make your code more
reproducible.</p>
</div>
<div id="set-file-locations-relative-to-the-project-folder-using-here"
class="section level3">
<h3>Set file locations relative to the project folder using
<code>here</code></h3>
<pre class="r"><code>here::i_am(&quot;analysis/Lab3_ProgrammingAlphaSimR.Rmd&quot;)</code></pre>
<p>There are two functions I use in the <code>here</code> package:
<code>here::i_am</code> and <code>here::here</code>.
<code>here::i_am</code> helps <code>here</code> “understand” the folder
structure. You give it the name of the script that is currently running,
and the path to that script from the base of the overall analysis
folder. From that, <code>here</code> deduces where the base of the
folder is. After that, <code>here::here</code> locates files from that
folder base. If you use the <code>here</code> package, you can send
anyone a complicated zipped folder, and they should be able to run the
analysis out of the box, regardless of where they put the folder in
their own file system.</p>
</div>
<div id="document-packages-used" class="section level3">
<h3>Document packages used</h3>
<p>This chunk creates a “README.md” file that documents the packages and
versions used for future reference.</p>
<pre class="r"><code>source(here::here(&quot;code/addToREADME.R&quot;))
addToREADME(paste0(&quot;## &quot;, rmarkdown::metadata$title), append=F)
addToREADME(c(date(), &quot;&quot;))

packages_info &lt;- ip[packages_used, c(&quot;Package&quot;, &quot;Version&quot;, &quot;Built&quot;)]
addToREADME(c(&quot;The packages used in this script are:&quot;, &quot;Package, Version, Built&quot;))
apply(packages_info, 1, function(vec) addToREADME(paste(vec, collapse=&quot; &quot;)))</code></pre>
<pre><code>              AlphaSimR               tidyverse               workflowr 
&quot;AlphaSimR 1.6.1 4.4.1&quot; &quot;tidyverse 2.0.0 4.4.0&quot; &quot;workflowr 1.7.1 4.4.0&quot; 
                   here 
     &quot;here 1.0.1 4.4.1&quot; </code></pre>
<pre class="r"><code>addToREADME(&quot;&quot;)</code></pre>
<p>Hyperlink the <a href="README.md">README file</a> to the report. That
makes it easy to find.</p>
</div>
<div id="set-random-seed" class="section level3">
<h3>Set random seed</h3>
<p>AlphaSimR generates many random numbers (e.g., to simulate Mendelian
random segregation). If you want the result of the analysis to come out
the same each time (there are pros and cons) you need to set the random
seed. Note that <code>workflowr</code> does this by default. If you are
not using that package, then be explicit.</p>
<pre class="r"><code>random_seed &lt;- 45678
set.seed(random_seed)
addToREADME(c(paste(&quot;The random seed is&quot;, random_seed), &quot;&quot;))</code></pre>
</div>
<div id="script-parameters" class="section level3">
<h3>Script parameters</h3>
<p>If the behavior of your script depends on parameters that you set,
initialize them early on.</p>
<pre class="r"><code>nFounders &lt;- 100
nChr &lt;- 10 # Number of chromosomes
nSitesPerChr &lt;- 1000 # Number of segregating sites _per chromosome_
nQTLperChr &lt;- 10 # Vary this parameter to get oligo- versus poly- genic traits
nF1s &lt;- 200 # We are going to make F1s to test outbreak of variation
nF2s &lt;- 200 # We are going to make F2s to test outbreak of variation</code></pre>
</div>
<div id="parameters-to-readme" class="section level3">
<h3>Parameters to README</h3>
<p>It’s good to have all the information together in the README. Chunks
of code like this do not need to be included in the report. To exclude
them use the “include=FALSE” option in the chunk.</p>
</div>
</div>
<div id="simulating-some-classical-results" class="section level1">
<h1>Simulating some classical results</h1>
<p>This script uses <code>AlphaSimR</code> to simulate the “outbreak of
variation” that arises when you self-fertilize a hybrid.</p>
<div id="alphasimr-populations" class="section level3">
<h3>AlphaSimR populations</h3>
<p>The basic object of <code>AlphaSimR</code> is the population. To make
founders, you first make founder haplotypes from a coalescent
simulation, then you define simulation parameters that will link their
genetic variation to phenotypic variation, then you make a first diploid
population from the founder haplotypes.</p>
<pre class="r"><code># Create haplotypes for founder population of outbred individuals
# Note: default effective population size for runMacs is 100
founderHaps &lt;- AlphaSimR::runMacs(nInd=nFounders, nChr=nChr, 
                                  segSites=nSitesPerChr)
# founderHaps &lt;- AlphaSimR::quickHaplo(nInd=nFounders, nChr=nChr, 
#                                  segSites=nSitesPerChr)

# New global simulation parameters from founder haplotypes
SP &lt;- AlphaSimR::SimParam$new(founderHaps)
# Additive trait architecture
# By default, the genetic variance will be 1
SP$addTraitA(nQtlPerChr=nQTLperChr)

# Create a new population of founders
founders &lt;- AlphaSimR::newPop(founderHaps, simParam=SP)
str(founders)</code></pre>
<pre><code>Formal class &#39;Pop&#39; [package &quot;AlphaSimR&quot;] with 18 slots
  ..@ id     : chr [1:100] &quot;1&quot; &quot;2&quot; &quot;3&quot; &quot;4&quot; ...
  ..@ iid    : int [1:100] 1 2 3 4 5 6 7 8 9 10 ...
  ..@ mother : chr [1:100] &quot;0&quot; &quot;0&quot; &quot;0&quot; &quot;0&quot; ...
  ..@ father : chr [1:100] &quot;0&quot; &quot;0&quot; &quot;0&quot; &quot;0&quot; ...
  ..@ sex    : chr [1:100] &quot;H&quot; &quot;H&quot; &quot;H&quot; &quot;H&quot; ...
  ..@ nTraits: int 1
  ..@ gv     : num [1:100, 1] -0.931 0.352 0.958 2.043 0.547 ...
  .. ..- attr(*, &quot;dimnames&quot;)=List of 2
  .. .. ..$ : NULL
  .. .. ..$ : chr &quot;Trait1&quot;
  ..@ pheno  : num [1:100, 1] NA NA NA NA NA NA NA NA NA NA ...
  .. ..- attr(*, &quot;dimnames&quot;)=List of 2
  .. .. ..$ : NULL
  .. .. ..$ : chr &quot;Trait1&quot;
  ..@ ebv    : num[1:100, 0 ] 
  ..@ gxe    :List of 1
  .. ..$ : NULL
  ..@ fixEff : int [1:100] 1 1 1 1 1 1 1 1 1 1 ...
  ..@ misc   : list()
  ..@ miscPop: list()
  ..@ nInd   : int 100
  ..@ nChr   : int 10
  ..@ ploidy : int 2
  ..@ nLoci  : int [1:10] 1000 1000 1000 1000 1000 1000 1000 1000 1000 1000
  ..@ geno   :List of 10
  .. ..$ : raw [1:125, 1:2, 1:100] 59 c4 b1 d2 ...
  .. ..$ : raw [1:125, 1:2, 1:100] 05 f8 8a 22 ...
  .. ..$ : raw [1:125, 1:2, 1:100] 56 a0 38 2a ...
  .. ..$ : raw [1:125, 1:2, 1:100] 05 52 24 a1 ...
  .. ..$ : raw [1:125, 1:2, 1:100] 23 70 84 b6 ...
  .. ..$ : raw [1:125, 1:2, 1:100] 1f 89 e1 90 ...
  .. ..$ : raw [1:125, 1:2, 1:100] 2c 2c 24 81 ...
  .. ..$ : raw [1:125, 1:2, 1:100] 06 90 89 78 ...
  .. ..$ : raw [1:125, 1:2, 1:100] 55 04 cb 13 ...
  .. ..$ : raw [1:125, 1:2, 1:100] ac 46 a1 2d ...</code></pre>
</div>
<div id="population-information" class="section level3">
<h3>Population information</h3>
<p>The population has <code>id</code>s. The <code>@mother</code> and
<code>@father</code> ids are all zero because this population was made
from founder haplotypes, and so does not have diploid parents. The
genotypic values <code>gv</code> of the population are calculated for
the trait created using
<code>SP$addTraitA(nQtlPerChr=nQTLperChr)</code>. Given that there are
10 chromosomes and 10 QTL per chromosome, there are
<code>3^(nChr*nQTLperChr)</code> = 5.1537752^{47} different possible
genotypic values. The realized genotypic values are accessible with the
function <code>gv(founders)</code></p>
<p>From here, you can treat this population like a named vector using
the square braces extraction operator <code>[ ]</code>. Extract
individuals by their <code>@id</code> or just by their order in the
population using an integer index. For example, pick three random
individuals from a population and list their ids. Pick the one with the
first id in alphabetical order.</p>
<pre class="r"><code>test &lt;- founders[c(2, 3, 5, 8, 13)]
testID &lt;- test@id
firstID &lt;- testID[order(testID)][1]
alphaInd &lt;- test[firstID] # Extract individual with the first ID in order
print(testID)</code></pre>
<pre><code>[1] &quot;2&quot;  &quot;3&quot;  &quot;5&quot;  &quot;8&quot;  &quot;13&quot;</code></pre>
<pre class="r"><code>print(alphaInd)</code></pre>
<pre><code>An object of class &quot;Pop&quot; 
Ploidy: 2 
Individuals: 1 
Chromosomes: 10 
Loci: 10000 
Traits: 1 </code></pre>
<pre class="r"><code>print(alphaInd@id)</code></pre>
<pre><code>[1] &quot;13&quot;</code></pre>
</div>
<div id="outbreak-of-variation" class="section level3">
<h3>Outbreak of variation</h3>
<p>Emerson and East (1913) showed that if you crossed two inbreds, the
hybrid had similar variation to each inbred, but if you then selfed the
hybrid, the offspring varied more substantially. This code simulates
that result. First, self the founders to homozygosity. The function
<code>self</code> self-fertilizes individuals from the population. By
default, it creates one selfed individual per parent (controllable with
the parameter <code>nProgeny</code>), so this works nicely for
single-seed descent.</p>
<pre class="r"><code># Self-pollinate to for a few generations
nGenSelf &lt;- 3
inbredPop &lt;- founders
for (gen in 1:nGenSelf){
  inbredPop &lt;- AlphaSimR::self(inbredPop)
}</code></pre>
</div>
<div id="check-homozygosity" class="section level3">
<h3>Check homozygosity</h3>
<p>Just a sanity check that this has, in fact, created a population of
100 individuals that are appropriately homozygous. Loci are coded 0, 1,
2. So <code>qtl == 1</code> represents the case were a locus is
heterozygous. <code>sum(qtl == 1)</code> counts those cases.</p>
<pre class="r"><code>qtl &lt;- AlphaSimR::pullQtlGeno(inbredPop)
if (nrow(qtl) != nFounders) stop(&quot;The number of individuals is unexpected&quot;)
if (ncol(qtl) != nChr * nQTLperChr) stop(&quot;The number of QTL is unexpected&quot;)
fracHet &lt;- sum(qtl == 1) / (nFounders * nChr * nQTLperChr)
cat(&quot;Expected fraction heterozygous&quot;, 1 / 2^nGenSelf, &quot;\n&quot;,
    &quot;Observed fraction heterozygous&quot;, fracHet, &quot;+/-&quot;, 
    round(2*sqrt(fracHet*(1-fracHet)/(nFounders*nChr*nQTLperChr)), 3), &quot;\n&quot;)</code></pre>
<pre><code>Expected fraction heterozygous 0.125 
 Observed fraction heterozygous 0.0307 +/- 0.003 </code></pre>
<blockquote>
<p>What was wrong with my reasoning about the Expected fraction
heterozygous?</p>
</blockquote>
<p>Not all of the QTL loci started out as heterozygous in the
founders.</p>
</div>
<div id="simulate-outbreak-of-variation" class="section level3">
<h3>Simulate outbreak of variation</h3>
<p>We will<br />
1. pick a random pair of inbred individuals<br />
2. cross that pair<br />
3. find out the variation in genotypic value among the pair’s
progeny<br />
4. pick a random F1 progeny<br />
5. self-fertilize that F1<br />
6. find out the variation in genotypic value among the F1’s
progeny<br />
<em>It’s often useful to write very high-level outlines of what you want
your code to do before you write the code</em><br />
We will assume a trait that has a heritability of 0.5 in the base,
non-inbred population. An easy way to get that heritability is with
genetic and error variances of 1.</p>
</div>
<div id="pick-random-pair-to-make-f1-hybrid" class="section level3">
<h3>Pick random pair to make F1 hybrid</h3>
<pre class="r"><code>randomPair &lt;- inbredPop[sample(nFounders, 2)]</code></pre>
</div>
<div id="cross-the-pair-to-make-population-of-f1s"
class="section level3">
<h3>Cross the pair to make population of F1s</h3>
<p>The crossPlan is a <strong>two-column matrix</strong> with as many
rows as the number of crosses you want to make, the first column is the
<code>@id</code> or the index of the seed parent, and likewise for the
pollen parent in the second column. We will make 200 F1s from this
random pair, so the matrix has 200 rows. You just want to cross
individual 1 with individual 2, so each row contains 1:2.</p>
<pre class="r"><code>crossPlan &lt;- matrix(rep(1:2, nF1s), ncol=2, byrow=T)
f1_pop &lt;- AlphaSimR::makeCross(randomPair, crossPlan)
head(crossPlan)</code></pre>
<pre><code>     [,1] [,2]
[1,]    1    2
[2,]    1    2
[3,]    1    2
[4,]    1    2
[5,]    1    2
[6,]    1    2</code></pre>
</div>
<div id="find-out-the-phenotypic-variation-among-f1s"
class="section level3">
<h3>Find out the phenotypic variation among F1s</h3>
<p>When you first make a population, <code>AlphaSimR</code> does not
assume it has been phenotyped. You can phenotype it using the
<code>setPheno</code> function. Note that if you use
<code>setPheno</code> on the same population a second time, that will
overwrite the phenotypes from the first time. The genotypic variance can
be retrieved using the function <code>varG</code>. Really,
<code>varG</code> gives all we need to know, but of course that variance
is generally not observable in non-simulated reality.</p>
<pre class="r"><code>f1_pop &lt;- AlphaSimR::setPheno(f1_pop, varE=1)
cat(&quot;Genotypic variance among F1s&quot;, round(AlphaSimR::varG(f1_pop), 3), &quot;\n&quot;)</code></pre>
<pre><code>Genotypic variance among F1s 0.018 </code></pre>
<pre class="r"><code>cat(&quot;Phenotypic variance among F1s&quot;, round(AlphaSimR::varP(f1_pop), 3), &quot;\n&quot;)</code></pre>
<pre><code>Phenotypic variance among F1s 0.936 </code></pre>
<pre class="r"><code>hist(AlphaSimR::pheno(f1_pop), main=&quot;Histogram of F1 Phenotypes&quot;)</code></pre>
<p><img src="figure/Lab3_ProgrammingAlphaSimR.Rmd/Phenotype%20F1s-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="pick-random-f1" class="section level3">
<h3>Pick random F1</h3>
<pre class="r"><code>randomF1 &lt;- f1_pop[sample(nFounders, 1)]</code></pre>
</div>
<div id="make-f2-and-observe-phenotypic-variance"
class="section level3">
<h3>Make F2 and observe phenotypic variance</h3>
<p>When you first make a population, <code>AlphaSimR</code> does not
assume it has been phenotyped. You can phenotype it using the
<code>setPheno</code> function. Note that if you use
<code>setPheno</code> on the same population a second time, that will
overwrite the phenotypes from the first time. The genotypic variance can
be retrieved using the function <code>varG</code>. Really,
<code>varG</code> gives all we need to know, but of course that variance
is generally not observable in non-simulated reality.</p>
<pre class="r"><code>f2_pop &lt;- AlphaSimR::self(randomF1, nProgeny=nF2s)
f2_pop &lt;- AlphaSimR::setPheno(f2_pop, varE=1)
cat(&quot;Genotypic variance among F2s&quot;, round(AlphaSimR::varG(f2_pop), 3), &quot;\n&quot;)</code></pre>
<pre><code>Genotypic variance among F2s 0.412 </code></pre>
<pre class="r"><code>cat(&quot;Phenotypic variance among F2s&quot;, round(AlphaSimR::varP(f2_pop), 3), &quot;\n&quot;)</code></pre>
<pre><code>Phenotypic variance among F2s 1.373 </code></pre>
<pre class="r"><code>hist(AlphaSimR::pheno(f2_pop), main=&quot;Histogram of F2 Phenotypes&quot;)</code></pre>
<p><img src="figure/Lab3_ProgrammingAlphaSimR.Rmd/Make%20F2-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="illustrate-herehere-by-writing-a-file" class="section level3">
<h3>Illustrate <code>here::here</code> by writing a file</h3>
<p>Just to practice again using the here package because it makes life
easy.</p>
<pre class="r"><code>utils::write.csv(qtl, here::here(&quot;output&quot;, &quot;QTLgenotypes.csv&quot;))</code></pre>
</div>
</div>
<div id="homework" class="section level1">
<h1>Homework</h1>
<p>Illustrate regression to the mean between parents and offspring using
AlphaSimR<br />
1. You know how to create a population – create a population of
parents</p>
<p>I am going to try using the inbred=TRUE argument in runMacs() to see
the impact this has.</p>
<pre class="r"><code>set.seed(1234)

nFounders_b &lt;- 200
nChr &lt;- 10
nSitesPerChr &lt;- 1000
parents_haps &lt;- AlphaSimR::runMacs(nFounders_b, nChr, nSitesPerChr, inbred = FALSE)
SP_b &lt;- AlphaSimR::SimParam$new(parents_haps)
SP_b$addTraitA(nQtlPerChr = 100)
parents &lt;- AlphaSimR::newPop(parents_haps,SP_b)</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>You know how to get phenotypes from that population</li>
</ol>
<pre class="r"><code>par_pop &lt;- AlphaSimR::setPheno(parents, varE=1,simParam = SP_b)
hist(par_pop@pheno, main=&quot;Parent Phenotypes&quot;, xlab = &quot;Phenotype&quot;)</code></pre>
<p><img src="figure/Lab3_ProgrammingAlphaSimR.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<ol start="3" style="list-style-type: decimal">
<li>You want to randomly mate that population to get progeny that will
be (we assume) regressed to the mean. Create a <code>crossPlan</code>
matrix like I did to generate the F1s, except that each row should have
randomly-picked parents, rather than 1 and 2 like for the F1s. There is
also a command <code>AlphaSimR::randCross</code>. Check out its
documentation. If you use that command, you will have to find the
pedigree of the progeny using the <code>@mother</code> and
<code>@father</code> ids of the progeny population and match those up to
the parent population. If you make the crossPlan, then it gives you the
seed and pollen parent ids.</li>
</ol>
<pre class="r"><code>nF1s_b &lt;- 200
pollen_parents &lt;- sample(min(par_pop@iid):max(par_pop@iid), nF1s_b, replace = TRUE)
seed_parents &lt;- sample(min(par_pop@iid):max(par_pop@iid), nF1s_b, replace = TRUE)
crossPlan_b &lt;- as.matrix(cbind(pollen_parents,seed_parents))

f1_pop_b &lt;- AlphaSimR::makeCross(par_pop,crossPlan = crossPlan_b, simParam = SP_b)</code></pre>
<ol start="4" style="list-style-type: decimal">
<li>Having made the progeny population, phenotype it also</li>
</ol>
<pre class="r"><code>f1_pop_b_pheno &lt;- AlphaSimR::setPheno(f1_pop_b, varE=1, simParam = SP_b)
hist(f1_pop_b_pheno@pheno, main=&quot;F1 Phenotypes&quot;, xlab=&quot;Phenotypes&quot;)</code></pre>
<p><img src="figure/Lab3_ProgrammingAlphaSimR.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<ol start="5" style="list-style-type: decimal">
<li>Use each row of the <code>crossPlan</code> to find the two parents
and calculate their phenotypic mean</li>
</ol>
<pre class="r"><code>crossPlan_b &lt;- as.data.frame(crossPlan_b)
colnames(crossPlan_b)=c(&quot;pollen_parent&quot;,&quot;seed_parent&quot;)
crossPlan_b$seed_parent_pheno &lt;- NA
crossPlan_b$pollen_parent_pheno &lt;- NA
for (i in 1:nrow(crossPlan_b)){
  seed_parent &lt;- par_pop[crossPlan_b[i,&quot;seed_parent&quot;]]
  pollen_parent &lt;- par_pop[crossPlan_b[i,&quot;pollen_parent&quot;]]
  crossPlan_b[i,&quot;seed_parent_pheno&quot;] &lt;- seed_parent@pheno
  crossPlan_b[i,&quot;pollen_parent_pheno&quot;] &lt;- pollen_parent@pheno
}
crossPlan_b$midparent_value &lt;- rowMeans(cbind(crossPlan_b$seed_parent_pheno,crossPlan_b$pollen_parent_pheno))

phenotypes &lt;- as.data.frame(cbind(crossPlan_b$midparent_value, f1_pop_b_pheno@pheno))
colnames(phenotypes) &lt;- c(&quot;midparent_value&quot;, &quot;progeny_pheno&quot;)</code></pre>
<ol start="6" style="list-style-type: decimal">
<li>Make a scatterplot of the progeny phenotypes against the parent mean
phenotypes</li>
</ol>
<pre class="r"><code>ggplot(data=phenotypes, aes(x=midparent_value, y=progeny_pheno)) +
  geom_point()</code></pre>
<p><img src="figure/Lab3_ProgrammingAlphaSimR.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<blockquote>
<p>What is the regression coefficient?</p>
</blockquote>
<pre class="r"><code>lin.mod &lt;- lm(progeny_pheno~midparent_value, data=phenotypes)
reg_coef &lt;- lin.mod$coefficients[2]
print(reg_coef)</code></pre>
<pre><code>midparent_value 
      0.5256184 </code></pre>
<p>The regression coefficient of progeny phenotypes regressed on
midparent values is 0.4784871.</p>
<blockquote>
<p>How much closer to the mean are offspring, on average, compared to
their parents?</p>
</blockquote>
<pre class="r"><code>mean_pheno &lt;- mean(c(phenotypes$progeny_pheno,par_pop@pheno))
print(mean_pheno)</code></pre>
<pre><code>[1] 0.05633017</code></pre>
<pre class="r"><code>phenotypes$midpar_dev &lt;- phenotypes$midparent_value-mean_pheno
avg_midpar_dev &lt;- mean(phenotypes$midpar_dev)
phenotypes$prog_dev &lt;- phenotypes$progeny_pheno-mean_pheno
avg_prog_dev &lt;- mean(phenotypes$prog_dev)
avg_midpar_dev-avg_prog_dev</code></pre>
<pre><code>[1] -0.03205851</code></pre>
<p>On average, progeny were 0.1455552 units closer to the average
phenotype in the population than their parents.</p>
<div id="homework-grading" class="section level3">
<h3>Homework grading</h3>
<p>I want to receive a zipped folder with your homework in it one week
from today, by midnight. If you use the <code>workflowr</code> package
to create the folder structure, that’s great. You should have created an
Rmarkdown script that will create an html file that I can look through.
So:<br />
1. 4 points for just sending in a zipped folder on time<br />
2. 1 points for each successful step from above, with partial
credit<br />
3. I will subtract 0.5 points for every day the homework is late</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: aarch64-apple-darwin20
Running under: macOS 15.3.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] lubridate_1.9.4 forcats_1.0.0   stringr_1.5.1   dplyr_1.1.4    
 [5] purrr_1.0.2     readr_2.1.5     tidyr_1.3.1     tibble_3.2.1   
 [9] ggplot2_3.5.1   tidyverse_2.0.0

loaded via a namespace (and not attached):
 [1] sass_0.4.9        generics_0.1.3    stringi_1.8.4     hms_1.1.3        
 [5] digest_0.6.37     magrittr_2.0.3    evaluate_1.0.1    grid_4.4.1       
 [9] timechange_0.3.0  fastmap_1.2.0     rprojroot_2.0.4   workflowr_1.7.1  
[13] jsonlite_1.8.9    whisker_0.4.1     promises_1.3.2    scales_1.3.0     
[17] jquerylib_0.1.4   Rdpack_2.6.2      cli_3.6.3         crayon_1.5.3     
[21] rlang_1.1.4       rbibutils_2.3     bit64_4.5.2       munsell_0.5.1    
[25] withr_3.0.2       cachem_1.1.0      yaml_2.3.10       AlphaSimR_1.6.1  
[29] parallel_4.4.1    tools_4.4.1       tzdb_0.4.0        colorspace_2.1-1 
[33] httpuv_1.6.15     here_1.0.1        vctrs_0.6.5       R6_2.5.1         
[37] lifecycle_1.0.4   git2r_0.35.0      bit_4.5.0.1       fs_1.6.5         
[41] vroom_1.6.5       pkgconfig_2.0.3   pillar_1.10.0     bslib_0.8.0      
[45] later_1.4.1       gtable_0.3.6      glue_1.8.0        Rcpp_1.0.13-1    
[49] xfun_0.49         tidyselect_1.2.1  rstudioapi_0.17.1 knitr_1.49       
[53] farver_2.1.2      htmltools_0.5.8.1 labeling_0.4.3    rmarkdown_2.29   
[57] compiler_4.4.1   </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
