<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Kelly Robbins" />

<meta name="date" content="2025-02-18" />

<title>Benefits of Using Genetic Relationships</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PLBRG_7170</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Benefits of Using Genetic
Relationships</h1>
<h4 class="author">Kelly Robbins</h4>
<h4 class="date">2025-02-18</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-02-26
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>PLBRG_7170/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250128code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250128)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250128code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250128)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtmulderrigPLBRG7170treeba34ff682968aca86fb420fe14b17c214d91cb59targetblankba34ff6a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tmulderrig/PLBRG_7170/tree/ba34ff682968aca86fb420fe14b17c214d91cb59" target="_blank">ba34ff6</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtmulderrigPLBRG7170treeba34ff682968aca86fb420fe14b17c214d91cb59targetblankba34ff6a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/tmulderrig/PLBRG_7170/tree/ba34ff682968aca86fb420fe14b17c214d91cb59" target="_blank">ba34ff6</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store

Untracked files:
    Untracked:  analysis/Lab4_DataForBreedingValue.Rmd
    Untracked:  analysis/Lab_5.Rmd
    Untracked:  analysis/pset1_tpm.pdf
    Untracked:  analysis/pset2_tpm.Rmd
    Untracked:  code/Simulate_and_Solve.R
    Untracked:  data/Genomic_Rel.csv
    Untracked:  data/Pedigree.csv
    Untracked:  data/Pedigree_Rel.csv
    Untracked:  data/SNP_data.csv

Unstaged changes:
    Modified:   analysis/Lab3_ProgrammingAlphaSimR.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/lab2_MatrixAlgebra.Rmd
    Modified:   analysis/pset1_tpm.Rmd
    Modified:   output/QTLgenotypes.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="genetic-relationships-identity-by-descent"
class="section level2">
<h2>Genetic Relationships: Identity by Descent</h2>
<div
id="the-genetic-covariance-between-individuals-for-a-given-phenotype-is-determined-by-common-alleles-for-the-qtl-that-explain-variance-for-the-phenotype"
class="section level3">
<h3>The genetic (co)variance between individuals for a given phenotype
is determined by common alleles for the QTL that explain variance for
the phenotype</h3>
</div>
<div
id="in-the-absence-of-detailed-information-on-the-qtl-the-covariance-can-be-approximated-by-calculating-the-probability-two-individuals-carry-alleles-that-are-identical-by-descent-ibd."
class="section level3">
<h3>In the absence of detailed information on the QTL, the (co)variance
can be approximated by calculating the probability two individuals carry
alleles that are identical by descent (IBD).</h3>
</div>
<div
id="ibd-represents-the-probability-that-the-same-allele-is-inherited-by-two-individuals-from-some-common-ancestor."
class="section level3">
<h3>- IBD represents the probability that the same allele is inherited
by two individuals from some common ancestor.</h3>
</div>
<div id="these-calculations-can-be-done-using-pedigrees-or-genotypes"
class="section level3">
<h3>- These calculations can be done using pedigrees or genotypes</h3>
</div>
</div>
<div id="a-simple-illustration" class="section level2">
<h2>A simple illustration</h2>
<div
id="in-this-exmple-we-are-testing-lines-from-2-unrelated-full-sib-families.-the-code-below-will-simulate-a-polygenetic-trait-and-compare-prediction-accuracy-using-a-model-that-ignores-genetic-relationships-and-a-mixed-model-that-accounts-for-the-relationships.-you-will-need-to-download-and-install-the-package-mass"
class="section level3">
<h3>In this exmple we are testing lines from 2 unrelated full sib
families. The code below will simulate a polygenetic trait and compare
prediction accuracy using a model that ignores genetic relationships and
a mixed model that accounts for the relationships. <strong><em>you will
need to download and install the package MASS</em></strong></h3>
<p>##Setting up parameters</p>
<pre class="r"><code>library(MASS)
library(here)</code></pre>
<pre><code>here() starts at /Users/tm639/Documents/Github/PLBRG_7170</code></pre>
<pre class="r"><code>here::i_am(&quot;analysis/Lab_5.Rmd&quot;)</code></pre>
<pre><code>here() starts at /Users/tm639/Documents/Github/PLBRG_7170</code></pre>
<pre class="r"><code>#setting variety and residual variance
VarA=75
VarE=25
#mean of the families relative to the overall mean
meanFam=c(rep(-5,15),rep(5,15))</code></pre>
</div>
</div>
<div id="generating-relationship-matrix" class="section level2">
<h2>Generating relationship matrix</h2>
<pre class="r"><code>source(here::here(&quot;code/Simulate_and_Solve.R&quot;))
# this is the expected relationship matrix for to unrelated full-sib families
# there is a total of 30 lines, 15 lines in each family
#initializing a 30 x 30 matrix with all values = 0
A=matrix(0,30,30)
#creating a block of .5 for family 1
A[1:15,1:15]=.5
#creating a block of .5 for family 2
A[16:30,16:30]=.5
#setting the diagonal to 1
diag(A)&lt;-1
print(&quot;A matrix&quot;)</code></pre>
<pre><code>[1] &quot;A matrix&quot;</code></pre>
<pre class="r"><code>image(A)</code></pre>
<p><img src="figure/Lab_5.Rmd/Relationship%20Matirx-1.png" width="672" style="display: block; margin: auto;" />
## Simulate Breeding Values</p>
<pre class="r"><code>set.seed(35783948)
# sampling 30 variety effects
# now we are sampling from mvnorm since the variety effects are no longer iid
BV=MASS::mvrnorm(1,meanFam,VarA*A)

sol=simulate_and_solve(BV,A,VarA,VarE)
# LSM - Least squares means
# MM - BLUPs from the mixed model
sol</code></pre>
<pre><code>            LSM        MM
 [1,]  88.07296  89.49569
 [2,]  95.88636  95.35574
 [3,]  83.62355  86.15863
 [4,] 102.11244 100.02530
 [5,] 103.33145 100.93956
 [6,] 101.57676  99.62354
 [7,]  84.72632  86.98571
 [8,]  85.26471  87.38950
 [9,]  80.14345  83.54856
[10,] 103.85069 101.32899
[11,]  93.64864  93.67745
[12,]  95.57741  95.12403
[13,]  98.69429  97.46169
[14,]  98.20998  97.09846
[15,]  84.78968  87.03323
[16,] 105.81386 105.40746
[17,]  99.69822 100.82072
[18,]  85.53873  90.20111
[19,] 101.17771 101.93034
[20,] 109.05235 107.83632
[21,] 107.32753 106.54271
[22,]  96.53493  98.44825
[23,] 106.48451 105.91044
[24,]  99.41262 100.60652
[25,] 116.75780 113.61541
[26,] 109.33145 108.04564
[27,] 110.52786 108.94295
[28,] 102.91533 103.23355
[29,] 113.32767 111.04281
[30,] 105.87243 105.45138</code></pre>
</div>
<div id="questions" class="section level2">
<h2>Questions</h2>
<p>1)Conduct the following simulations using a difference in mean
performance between the two families of 10.</p>
<p>A.(1 pt) Use VarA = 75 and VarE = 25 (heritability = .75). Change the
seeds and run the simulation 5 times - Calculate the correlations
between LSM and the true BV - Calculate the correlations between MM and
the true BV - Plot the BLUPs against LSM</p>
<pre class="r"><code>corLSM_BV &lt;- c()
corMM_BV &lt;- c()

VarA=75
VarE=25
iterations &lt;- 5

starting_seed=1234
for(i in 1:iterations){
  set.seed(starting_seed+i)
  BV=MASS::mvrnorm(1,meanFam,VarA*A)
  solutions &lt;- simulate_and_solve(BV,A,VarA,VarE)
  corLSM_BV[i] &lt;- cor(solutions[,&quot;LSM&quot;],BV)
  corMM_BV[i] &lt;- cor(solutions[,&quot;MM&quot;],BV)
  plot(solutions[,&quot;MM&quot;],solutions[,&quot;LSM&quot;], main=&quot;BLUPs vs LSM estimates&quot;, sub = paste(&quot;Iteration:&quot;,i, sep = &quot;&quot;))
}</code></pre>
<p><img src="figure/Lab_5.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Lab_5.Rmd/unnamed-chunk-1-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Lab_5.Rmd/unnamed-chunk-1-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Lab_5.Rmd/unnamed-chunk-1-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Lab_5.Rmd/unnamed-chunk-1-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>mean(corLSM_BV)</code></pre>
<pre><code>[1] 0.9116239</code></pre>
<pre class="r"><code>mean(corMM_BV)</code></pre>
<pre><code>[1] 0.9214164</code></pre>
<p>B.(1 pt) Use VarA = 50 and VarE = 50 (heritability = .5). Change the
seeds and run the simulation 5 times - calculate the correlations
between LSM and the true BV - calculate the correlations between MM and
the true BV - Plot the BLUPs against LSM</p>
<pre class="r"><code>corLSM_BV_2 &lt;- c()
corMM_BV_2 &lt;- c()

VarA=50
VarE=50
iterations &lt;- 5

starting_seed=2234
for(i in 1:iterations){
  set.seed(starting_seed+i)
  BV=MASS::mvrnorm(1,meanFam,VarA*A)
  solutions &lt;- simulate_and_solve(BV,A,VarA,VarE)
  corLSM_BV_2[i] &lt;- cor(solutions[,&quot;LSM&quot;],BV)
  corMM_BV_2[i] &lt;- cor(solutions[,&quot;MM&quot;],BV)
  plot(solutions[,&quot;MM&quot;],solutions[,&quot;LSM&quot;], main=&quot;BLUPs vs LSM estimates&quot;, sub = paste(&quot;Iteration:&quot;,i, sep = &quot;&quot;))
}</code></pre>
<p><img src="figure/Lab_5.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Lab_5.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Lab_5.Rmd/unnamed-chunk-2-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Lab_5.Rmd/unnamed-chunk-2-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Lab_5.Rmd/unnamed-chunk-2-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>mean(corLSM_BV_2)</code></pre>
<pre><code>[1] 0.8644865</code></pre>
<pre class="r"><code>mean(corMM_BV_2)</code></pre>
<pre><code>[1] 0.8818306</code></pre>
<p>C.(1 pt) Use VarA = 25 and VarE = 75 (heritability = .25). Change the
seeds and run the simulation 5 times - Calculate the correlations
between LSM and the true BV - Calculate the correlations between MM and
the true BV - Plot the BLUPs against LSM</p>
<pre class="r"><code>corLSM_BV_3 &lt;- c()
corMM_BV_3 &lt;- c()

VarA=25
VarE=75
iterations &lt;- 5

starting_seed=3234
for(i in 1:iterations){
  set.seed(starting_seed+i)
  BV=MASS::mvrnorm(1,meanFam,VarA*A)
  solutions &lt;- simulate_and_solve(BV,A,VarA,VarE)
  corLSM_BV_3[i] &lt;- cor(solutions[,&quot;LSM&quot;],BV)
  corMM_BV_3[i] &lt;- cor(solutions[,&quot;MM&quot;],BV)
  plot(solutions[,&quot;MM&quot;],solutions[,&quot;LSM&quot;], main=&quot;BLUPs vs LSM estimates&quot;, sub = paste(&quot;Iteration:&quot;,i, sep = &quot;&quot;))
}</code></pre>
<p><img src="figure/Lab_5.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Lab_5.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Lab_5.Rmd/unnamed-chunk-3-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Lab_5.Rmd/unnamed-chunk-3-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Lab_5.Rmd/unnamed-chunk-3-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>mean(corLSM_BV_3)</code></pre>
<pre><code>[1] 0.6422445</code></pre>
<pre class="r"><code>mean(corMM_BV_3)</code></pre>
<pre><code>[1] 0.7843616</code></pre>
<p>D. (1 pt) What trends do you see in the plots of the BLUPS versus
LSM? What would be causing this pattern?</p>
<p>When the heritability is high, the distributions for the estimated
breeding values of the two full-sib families are near each other. As the
trait becomes less heritable, the block effects between the observations
has a larger impact on the estimates, which leads to these two
distributions emerging.</p>
<p>E. (1 pt) How does heritability impact the relative performance BLUPs
and LSM in terms of accuracy (correlations to the true BV)? Why?</p>
<p>The change in heritability had a larger relative impact on the LSM
estimates rather than the BLUP estimates. The mean LSM estimates are
based only on the average values and as the additive variance decreases
relative to error variance, these estimates are based on lots of noise.
The BLUP estimates use related individuals to inform estimates, which
can estimate through the noise of the error variance.</p>
<ol start="2" style="list-style-type: decimal">
<li>Change the difference in mean performance between families to
5.</li>
</ol>
<ol style="list-style-type: upper-alpha">
<li>(3 pts) Repeat the simulations in outlined in (1A-1C)</li>
</ol>
<pre class="r"><code>corLSM_BV &lt;- c()
corMM_BV &lt;- c()

VarA=75
VarE=25
iterations &lt;- 5

starting_seed=1234
for(i in 1:iterations){
  set.seed(starting_seed+i)
  BV=MASS::mvrnorm(1,meanFam,VarA*A)
  solutions &lt;- simulate_and_solve(BV,A,VarA,VarE)
  corLSM_BV[i] &lt;- cor(solutions[,&quot;LSM&quot;],BV)
  corMM_BV[i] &lt;- cor(solutions[,&quot;MM&quot;],BV)
  plot(solutions[,&quot;MM&quot;],solutions[,&quot;LSM&quot;], main=&quot;BLUPs vs LSM estimates&quot;, sub = paste(&quot;Iteration:&quot;,i, sep = &quot;&quot;))
}</code></pre>
<p><img src="figure/Lab_5.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Lab_5.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Lab_5.Rmd/unnamed-chunk-4-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Lab_5.Rmd/unnamed-chunk-4-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Lab_5.Rmd/unnamed-chunk-4-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>mean(corLSM_BV)</code></pre>
<pre><code>[1] 0.9211714</code></pre>
<pre class="r"><code>mean(corMM_BV)</code></pre>
<pre><code>[1] 0.9288661</code></pre>
<ol start="2" style="list-style-type: upper-alpha">
<li>(1 pt) How did the change in the difference in mean performance
between families impact the relative performance of BLUPs and LSM?
Why?</li>
</ol>
<p>(1 point for turning in the assignment on time)</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: aarch64-apple-darwin20
Running under: macOS 15.3.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] here_1.0.1  MASS_7.3-61

loaded via a namespace (and not attached):
 [1] vctrs_0.6.5       cli_3.6.3         knitr_1.49        rlang_1.1.4      
 [5] xfun_0.49         stringi_1.8.4     promises_1.3.2    jsonlite_1.8.9   
 [9] workflowr_1.7.1   glue_1.8.0        rprojroot_2.0.4   git2r_0.35.0     
[13] htmltools_0.5.8.1 httpuv_1.6.15     sass_0.4.9        rmarkdown_2.29   
[17] evaluate_1.0.1    jquerylib_0.1.4   tibble_3.2.1      fastmap_1.2.0    
[21] yaml_2.3.10       lifecycle_1.0.4   stringr_1.5.1     compiler_4.4.1   
[25] fs_1.6.5          Rcpp_1.0.13-1     pkgconfig_2.0.3   rstudioapi_0.17.1
[29] later_1.4.1       digest_0.6.37     R6_2.5.1          pillar_1.10.0    
[33] magrittr_2.0.3    bslib_0.8.0       tools_4.4.1       cachem_1.1.0     </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
