<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Timothy Mulderrig" />

<meta name="date" content="2025-02-03" />

<title>Pset 1</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PLBRG_7170</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Pset 1</h1>
<h4 class="author">Timothy Mulderrig</h4>
<h4 class="date">2025-02-03</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-02-03
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>PLBRG_7170/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250128code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250128)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250128code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250128)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongcb3e41b">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> cb3e41b
</a>
</p>
</div>
<div id="strongRepositoryversionstrongcb3e41b"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version cb3e41b.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store

Untracked files:
    Untracked:  analysis/pset1_tpm.Rmd

Unstaged changes:
    Modified:   analysis/index.Rmd
    Modified:   analysis/lab1_Simulate.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="q1" class="section level1">
<h1>Q1</h1>
<p>As a plant breeder I want to fix an allele that provides resistance
to an important crop disease. Unfortunately, I do not have the ability
to directly genotype the causative mutation,but I have identified a
genetic marker that is closely linked the causative mutation. Based on
mapping experiments the recombination frequency between the marker
(alleles: A &amp; a) and the QTL (alleles: Q – resistant &amp; q –
susceptible) is r = .25</p>
<p>I have crossed a homozygous resistant line to a homozygous
susceptible line to get an offspring with the genotype: [A/Q] [a/q]</p>
<p>I will self this individual to generate offspring segregating for
resistance.</p>
<div id="a" class="section level2">
<h2>1A)</h2>
<p>If I select only lines with the genotype AA, what expected proportion
of the selected lines that will be homozygous for the favorable QTL
(QQ)? (2 pts)</p>
<p>Frequency of possible gametes</p>
<pre class="r"><code>r=0.25
freq_AQ &lt;-  (1-r)*0.5
freq_aq &lt;- (1-r)*0.5
freq_Aq &lt;- r*(0.5)
freq_aQ &lt;- r*(0.5)</code></pre>
<p>Frequency of Genotype combinations in F2</p>
<pre class="r"><code>#multiply by 2 in cases where unique gametes could originate from the pallen or egg donor

AAQQ &lt;- freq_AQ*freq_AQ
AAQq &lt;- freq_AQ*freq_Aq*2
AAqq &lt;- freq_Aq*freq_Aq
AaQQ &lt;- freq_AQ*freq_aQ*2
AaQq &lt;- freq_AQ*freq_aq*2 + freq_Aq*freq_aQ*2
Aaqq &lt;- freq_Aq*freq_aq*2
aaQQ &lt;- freq_aQ*freq_aQ
aaQq &lt;- freq_aQ*freq_aq*2
aaqq &lt;- freq_aq*freq_aq

AAQQ+AAQq+AAqq+AaQQ+AaQq+Aaqq+aaQQ+aaQq+aaqq</code></pre>
<pre><code>[1] 1</code></pre>
<p>Selected genotypes</p>
<pre class="r"><code>sel_genos &lt;- AAQQ + AAQq + AAqq
QQ_sel &lt;- AAQQ / sel_genos
print(QQ_sel)</code></pre>
<pre><code>[1] 0.5625</code></pre>
<div
id="answer-the-expected-proportion-of-aa-selected-lines-that-will-be-homozygous-for-the-resistance-qtl-is-56.25."
class="section level3">
<h3>Answer: The expected proportion of AA selected lines that will be
homozygous for the resistance QTL is 56.25%.</h3>
</div>
</div>
</div>
<div id="q2" class="section level1">
<h1>Q2</h1>
<p>Similar to Q1 I have identified markers linked to a QTL of interest
and I now want to practice marker assisted selection to fix the
favorable allele in a population. In this scenario I have 2 markers
linked on either side of the QTL. Marker 1 (alleles A, a) has a
recombination frequency r =.4 with the QTL and Marker 2 (alleles B, b)
has a recombination frequency r =.3 with the QTL. As before I’ve crossed
a line that is homozygous for the beneficial allele (Q) and a line that
is homozygous for the unfavorable allele (q):</p>
<p>[A/Q/B] x [a/q/b]</p>
<p>I selfed the heterozygous line above to generate many offspring. The
QQ genotype has a value of a = 5 for the trait of interest, the
unfavorable genotype qq has a value of -a = -5 for the trait of
interest. The trait is additive (no dominance)</p>
<div id="a-1" class="section level2">
<h2>2A)</h2>
<p>As a breeder I want to fix the resistance allele. If I select only
lines with the genotype AABB, what is the expected proportion of the
selected lines that will be homozygous for the favorable QTL (QQ)? (2
pts)</p>
<p>Frequency of possible gametes</p>
<pre class="r"><code>r1 &lt;- 0.4
r2 &lt;- 0.3

freq_AQB &lt;- 0.5*(1-r1)*(1-r2)
freq_AQb &lt;- 0.5*(1-r1)*(r2)
freq_Aqb &lt;- 0.5*(r1)*(1-r2)
freq_AqB &lt;- 0.5*(r1)*(r2)
freq_aQB &lt;- 0.5*(r1)*(1-r2)
freq_aQb &lt;- 0.5*(r1)*(r2)
freq_aqB &lt;- 0.5*(1-r1)*(r2)
freq_aqb &lt;- 0.5*(1-r1)*(1-r2)

freq_AQB + freq_AQb + freq_Aqb + freq_AqB + freq_aQB + freq_aQb + freq_aqB + freq_aqb</code></pre>
<pre><code>[1] 1</code></pre>
<p>Frequency of genotypes</p>
<pre class="r"><code>AAQQBB &lt;- freq_AQB*freq_AQB
AAQQBb &lt;- freq_AQB*freq_AQb*2
AAQQbb &lt;- freq_AQb*freq_AQb
AAQqBB &lt;- freq_AQB*freq_AqB*2
AAQqBb &lt;- freq_AQB*freq_Aqb*2 + freq_AQb*freq_AqB*2
AAQqbb &lt;- freq_AQb*freq_Aqb*2
AAqqBB &lt;- freq_AqB*freq_AqB
AAqqBb &lt;- freq_AqB*freq_Aqb*2
AAqqbb &lt;- freq_Aqb*freq_Aqb
AaQQBB &lt;- freq_AQB*freq_aQB*2 
AaQQBb &lt;- freq_AQB*freq_aQb*2 + freq_AQb*freq_aQB*2
AaQQbb &lt;- freq_AQb*freq_aQb*2
AaQqBB &lt;- freq_AQB*freq_aqB*2 + freq_AqB*freq_aQB*2
AaQqBb &lt;- freq_AQB*freq_aqb*2 + freq_AQb*freq_aqB*2 + freq_Aqb*freq_aQB*2 + freq_aQb*freq_AqB*2
AaQqbb &lt;- freq_AQb*freq_aqb*2 + freq_Aqb*freq_aQb*2
AaqqBB &lt;- freq_AqB*freq_aqB*2
AaqqBb &lt;- freq_AqB*freq_aqb*2 + freq_aqB*freq_Aqb*2
Aaqqbb &lt;- freq_Aqb*freq_aqb*2
aaQQBB &lt;- freq_aQB*freq_aQB
aaQQBb &lt;- freq_aQB*freq_aQb*2
aaQQbb &lt;- freq_aQb*freq_aQb
aaQqBB &lt;- freq_aQB*freq_aqB*2
aaQqBb &lt;- freq_aQB*freq_aqb*2+freq_aQb*freq_aqB*2
aaQqbb &lt;- freq_aQb*freq_aqb*2
aaqqBB &lt;- freq_aqB*freq_aqB
aaqqBb &lt;- freq_aqB*freq_aqb*2
aaqqbb &lt;- freq_aqb*freq_aqb

sum(c(AAQQBB, AAQQBb, AAQQbb, AAQqBB, AAQqBb, AAQqbb, AAqqBB, AAqqBb, AAqqbb, 
  AaQQBB, AaQQBb, AaQQbb, AaQqBB, AaQqBb, AaQqbb, AaqqBB, AaqqBb, Aaqqbb, 
  aaQQBB, aaQQBb, aaQQbb, aaQqBB, aaQqBb, aaQqbb, aaqqBB, aaqqBb, aaqqbb))</code></pre>
<pre><code>[1] 1</code></pre>
<p>Selected genotypes</p>
<pre class="r"><code>sel_genos_b &lt;- AAQQBB + AAQqBB + AAqqBB
QQ_sel_b &lt;- AAQQBB / sel_genos_b
print(QQ_sel_b)</code></pre>
<pre><code>[1] 0.6049383</code></pre>
<div
id="answer-the-expected-proportion-of-selected-lines-that-will-be-homozygous-for-the-resistance-qtl-is-60.49"
class="section level3">
<h3>Answer: The expected proportion of selected lines that will be
homozygous for the resistance QTL is 60.49 %</h3>
</div>
</div>
<div id="b" class="section level2">
<h2>2B)</h2>
<p>What is the expected value for offspring with the genotype AABB? (3
pts)</p>
<pre class="r"><code>AABB_e &lt;- AAQQBB*5 + AAQqBB*0 + AAqqBB*-5
print(AABB_e)</code></pre>
<pre><code>[1] 0.2025</code></pre>
<div id="answer-the-expected-value-for-offspring-with-aabb-is-0.2025"
class="section level3">
<h3>Answer: The expected value for offspring with AABB is 0.2025</h3>
</div>
</div>
<div id="c" class="section level2">
<h2>2C)</h2>
<p>What is the expected value for lines with the genotype AA? (3
pts)</p>
<pre class="r"><code>AAQQBB*5 + AAQQBb*5 + AAQQbb*5 + AAQqBB*0 + AAQqBb*0 + AAQqbb*0 + AAqqBB*(-5) + AAqqBb*(-5) + AAqqbb*(-5)</code></pre>
<pre><code>[1] 0.25</code></pre>
<div id="answer-the-expected-value-for-lines-with-aa-is-0.25"
class="section level3">
<h3>Answer: The expected value for lines with AA is 0.25</h3>
</div>
</div>
<div id="d" class="section level2">
<h2>2D)</h2>
<p>What is the expected value for lines with the genotype BB? (3
pts)</p>
<pre class="r"><code>AAQQBB*5 + AaQQBB*5 + aaQQBB*5 + AAQqBB*0 + AaQqBB*0 + aaQqBB*0 + AAqqBB*(-5) + AaqqBB*(-5) + aaqqBB*(-5)</code></pre>
<pre><code>[1] 0.5</code></pre>
<div id="answer-the-expected-value-for-lines-with-bb-is-0.5"
class="section level3">
<h3>Answer: The expected value for lines with BB is 0.5</h3>
</div>
</div>
<div id="e" class="section level2">
<h2>2E)</h2>
<p>How close would a single marker need to be in order for marker
assisted selection to as effective and the two marker example above? (2
pts)</p>
<p>Assume there is a marker c that is linked to qtl q with a
recombination frequency of rc between c and q in the gametes
[C/Q][c/q].</p>
<p>The potential gametes appear with the frequencies: * [C/Q] =
0.5<em>(1-rc) </em> [C/q] = 0.5<em>(rc) </em> [c/Q] = 0.5<em>(rc) </em>
[c/q] = 0.5*(1-rc)</p>
<p>The resulting genotypes appear with the frequenies:</p>
<ul>
<li>CCQQ &lt;- freq_CQ*freq_CQ</li>
<li>CCQq &lt;- freq_CQ* freq_Cq*2</li>
<li>CCqq &lt;- freq_Cq*freq_Cq</li>
<li>CcQQ &lt;- freq_CQ* freq_cQ*2</li>
<li>CcQq &lt;- freq_CQ* freq_cq* 2 + freq_Cq* freq_cQ*2</li>
<li>Ccqq &lt;- freq_Cq* freq_cq*2</li>
<li>ccQQ &lt;- freq_cQ* freq_cQ</li>
<li>ccQq &lt;- freq_cQ* freq_cq*2</li>
<li>ccqq &lt;- freq_cq*freq_cq</li>
</ul>
<p>If we select only the lines with CC marker scores, the expected
genotypic value would be as follows:</p>
<ul>
<li>Expected Value = CCQQ* 5 + CCQq* 0 + CCqq* -5 or</li>
<li>Expected Value = (0.5* (1-rc))* (0.5* (1-rc))* 5 + (0.5* rc)(0.5*
rc)* (-5)</li>
<li>Expected Value = (5)(0.25rc^2 -0.5rc+0.25) + (-5)(0.25rc^2)</li>
<li>Expected Value = 1.25rc^2 -2.5rc+1.25-1.25rc^2</li>
<li>Expected Value = -2.5rc+1.25</li>
</ul>
<pre class="r"><code>rc &lt;- (AABB_e-1.25) / (-2.5)
print(rc)</code></pre>
<pre><code>[1] 0.419</code></pre>
<div
id="answer-a-marker-would-need-to-have-a-recombination-frequency-between-itself-and-the-qtl-of-0.419-for-the-selected-lines-to-have-the-same-expected-value-as-the-lines-selected-with-our-flanking-markers"
class="section level3">
<h3>Answer: A marker would need to have a recombination frequency
between itself and the QTL of 0.419 for the selected lines to have the
same expected value as the lines selected with our flanking markers</h3>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: aarch64-apple-darwin20
Running under: macOS 15.3

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] vctrs_0.6.5       cli_3.6.3         knitr_1.49        rlang_1.1.4      
 [5] xfun_0.49         stringi_1.8.4     promises_1.3.2    jsonlite_1.8.9   
 [9] workflowr_1.7.1   glue_1.8.0        rprojroot_2.0.4   git2r_0.35.0     
[13] htmltools_0.5.8.1 httpuv_1.6.15     sass_0.4.9        rmarkdown_2.29   
[17] evaluate_1.0.1    jquerylib_0.1.4   tibble_3.2.1      fastmap_1.2.0    
[21] yaml_2.3.10       lifecycle_1.0.4   stringr_1.5.1     compiler_4.4.1   
[25] fs_1.6.5          Rcpp_1.0.13-1     pkgconfig_2.0.3   rstudioapi_0.17.1
[29] later_1.4.1       digest_0.6.37     R6_2.5.1          pillar_1.10.0    
[33] magrittr_2.0.3    bslib_0.8.0       tools_4.4.1       cachem_1.1.0     </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
