<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Timothy Mulderrig" />

<meta name="date" content="2025-02-26" />

<title>pset2_tpm</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PLBRG_7170</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">pset2_tpm</h1>
<h4 class="author">Timothy Mulderrig</h4>
<h4 class="date">2025-02-26</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-02-26
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>PLBRG_7170/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250128code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250128)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250128code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250128)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtmulderrigPLBRG7170treeba34ff682968aca86fb420fe14b17c214d91cb59targetblankba34ff6a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tmulderrig/PLBRG_7170/tree/ba34ff682968aca86fb420fe14b17c214d91cb59" target="_blank">ba34ff6</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtmulderrigPLBRG7170treeba34ff682968aca86fb420fe14b17c214d91cb59targetblankba34ff6a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/tmulderrig/PLBRG_7170/tree/ba34ff682968aca86fb420fe14b17c214d91cb59" target="_blank">ba34ff6</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store

Untracked files:
    Untracked:  analysis/Lab4_DataForBreedingValue.Rmd
    Untracked:  analysis/Lab_5.Rmd
    Untracked:  analysis/pset1_tpm.pdf
    Untracked:  analysis/pset2_tpm.Rmd
    Untracked:  code/Simulate_and_Solve.R
    Untracked:  data/Genomic_Rel.csv
    Untracked:  data/Pedigree.csv
    Untracked:  data/Pedigree_Rel.csv
    Untracked:  data/SNP_data.csv

Unstaged changes:
    Modified:   analysis/Lab3_ProgrammingAlphaSimR.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/lab2_MatrixAlgebra.Rmd
    Modified:   analysis/pset1_tpm.Rmd
    Modified:   output/QTLgenotypes.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="load-libraries-and-set-directory" class="section level2">
<h2>Load Libraries and set directory</h2>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.4     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<pre class="r"><code>library(here)</code></pre>
<pre><code>here() starts at /Users/tm639/Documents/Github/PLBRG_7170</code></pre>
<pre class="r"><code>here::i_am(&quot;analysis/pset2_tpm.Rmd&quot;)</code></pre>
<pre><code>here() starts at /Users/tm639/Documents/Github/PLBRG_7170</code></pre>
</div>
<div id="question-1" class="section level2">
<h2>Question 1</h2>
<p>Read in the file “Pedigree.csv” and write an R function to calculate
the additive relationships (first calculate the coefficients of
coancestry and then multiply by 2 to get the additive relationships) (5
pts)</p>
<pre class="r"><code>ped_data &lt;- read_csv(here::here(&quot;data/Pedigree.csv&quot;))</code></pre>
<pre><code>New names:
Rows: 40 Columns: 4
── Column specification
──────────────────────────────────────────────────────── Delimiter: &quot;,&quot; dbl
(4): ...1, id, seedPar, pollenPar
ℹ Use `spec()` to retrieve the full column specification for this data. ℹ
Specify the column types or set `show_col_types = FALSE` to quiet this message.
• `` -&gt; `...1`</code></pre>
<pre class="r"><code>str(ped_data)</code></pre>
<pre><code>spc_tbl_ [40 × 4] (S3: spec_tbl_df/tbl_df/tbl/data.frame)
 $ ...1     : num [1:40] 1 2 3 4 5 6 7 8 9 10 ...
 $ id       : num [1:40] 1 2 3 4 5 6 7 8 9 10 ...
 $ seedPar  : num [1:40] 0 0 0 0 0 0 0 0 0 0 ...
 $ pollenPar: num [1:40] 0 0 0 0 0 0 0 0 0 0 ...
 - attr(*, &quot;spec&quot;)=
  .. cols(
  ..   ...1 = col_double(),
  ..   id = col_double(),
  ..   seedPar = col_double(),
  ..   pollenPar = col_double()
  .. )
 - attr(*, &quot;problems&quot;)=&lt;externalptr&gt; </code></pre>
<pre class="r"><code>coef_coanc_mat &lt;- matrix(data=NA,nrow=nrow(ped_data), ncol = nrow(ped_data))

founders &lt;- ped_data$id[which(ped_data$seedPar==0 &amp; ped_data$pollenPar==0)]
coef_coanc_mat[1:length(founders),1:length(founders)]=0
diag(coef_coanc_mat)[founders]=0.5

for (i in (length(founders) + 1):nrow(ped_data)){
  for (j in 1:(i - 1)){
    i_seedpar &lt;- as.numeric(ped_data[i,&quot;seedPar&quot;, drop=TRUE])
    i_pollenpar &lt;- as.numeric(ped_data[i,&quot;pollenPar&quot;, drop=TRUE])
    fdv &lt;- coef_coanc_mat[i_pollenpar,j]
    fsv &lt;- coef_coanc_mat[i_seedpar,j]
    fuv &lt;- 1/2*(fdv+fsv)
    coef_coanc_mat[i,j]=fuv
    coef_coanc_mat[j,i]=fuv
  }
  i_seedpar &lt;- as.numeric(ped_data[i,&quot;seedPar&quot;, drop=TRUE])
  i_pollenpar &lt;- as.numeric(ped_data[i,&quot;pollenPar&quot;, drop=TRUE])
  fsd &lt;- coef_coanc_mat[i_seedpar,i_pollenpar]
  coef_coanc_mat[i,i]= 1/2*(1+fsd)
}

a_mat &lt;- coef_coanc_mat*2</code></pre>
<p>Check accuracy of a mat compared to provided pedigree matrix</p>
<pre class="r"><code>ped_mat &lt;- as.matrix(read_csv(here::here(&quot;data/Pedigree_Rel.csv&quot;)))</code></pre>
<pre><code>New names:
Rows: 40 Columns: 41
── Column specification
──────────────────────────────────────────────────────── Delimiter: &quot;,&quot; dbl
(41): ...1, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 1...
ℹ Use `spec()` to retrieve the full column specification for this data. ℹ
Specify the column types or set `show_col_types = FALSE` to quiet this message.
• `` -&gt; `...1`</code></pre>
<pre class="r"><code>ped_mat &lt;- ped_mat[,2:41]
  
sum(a_mat-ped_mat)</code></pre>
<pre><code>[1] 0</code></pre>
<p>These matrices are the same!</p>
</div>
<div id="question-2" class="section level2">
<h2>Question 2</h2>
<p>Read in the file “SNP_data.csv” and write an R function to calculate
the genomic relationship matrix. (5 pts)</p>
<pre class="r"><code>snp_data &lt;- read_csv(here::here(&quot;data/SNP_data.csv&quot;))</code></pre>
<pre><code>New names:
Rows: 40 Columns: 9801
── Column specification
──────────────────────────────────────────────────────── Delimiter: &quot;,&quot; dbl
(9801): ...1, 1_4, 1_6, 1_7, 1_9, 1_10, 1_12, 1_14, 1_15, 1_16, 1_18, 1_...
ℹ Use `spec()` to retrieve the full column specification for this data. ℹ
Specify the column types or set `show_col_types = FALSE` to quiet this message.
• `` -&gt; `...1`</code></pre>
<pre class="r"><code>snp_data &lt;- snp_data[,2:ncol(snp_data)]
## Center marker scores on each marker mean
z &lt;- as.matrix(apply(snp_data, 2, function(col) col-mean(col)))
## Calculate p and 1-p for each marker
MAF &lt;- function(snp_mat){
  pm &lt;- c()
  for (i in 1:ncol(snp_mat)){
    ref_allele &lt;- (sum(snp_mat[,i]==0)*2+sum(snp_mat[,i]==1))/(nrow(snp_mat)*2)
    alt_allele &lt;- (sum(snp_mat[,i]==2)*2+sum(snp_mat[,i]==1))/(nrow(snp_mat)*2)
    pm[i] &lt;- min(c(ref_allele,alt_allele))
  }
  return(pm)
}

p &lt;- MAF(snp_data)
q &lt;- 1-p

scaling_factor &lt;- 2*sum(p*q)
  
G &lt;- z%*%t(z) / scaling_factor
G &lt;- round(G,9)</code></pre>
<p>Check accuracy of g mat compared to provided GRM matrix</p>
<pre class="r"><code>grm &lt;- as.matrix(read_csv(here::here(&quot;data/Genomic_Rel.csv&quot;)))</code></pre>
<pre><code>New names:
Rows: 40 Columns: 41
── Column specification
──────────────────────────────────────────────────────── Delimiter: &quot;,&quot; dbl
(41): ...1, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 1...
ℹ Use `spec()` to retrieve the full column specification for this data. ℹ
Specify the column types or set `show_col_types = FALSE` to quiet this message.
• `` -&gt; `...1`</code></pre>
<pre class="r"><code>grm &lt;- grm[,2:ncol(grm)]
grm &lt;- round(grm,9)

sum(G-grm)</code></pre>
<pre><code>[1] 0</code></pre>
<p>These matrices are the same!</p>
</div>
<div id="question-3" class="section level2">
<h2>Question 3</h2>
<p>Plot the off diagonals of the pedigree relationship matrix (x-axis)
and the genomic relationship matrix (y-axis). Identify at least 1
difference and provide an explanation (2 pts)</p>
<pre class="r"><code>offDiag &lt;- function(relMat){
return(c(relMat[upper.tri(relMat)]))
}
plotting_data &lt;- data.frame(x=offDiag(a_mat), y=offDiag(G))
ggplot(data=plotting_data,aes(x=x, y=y))+
  geom_point(color=&quot;darkgreen&quot;) +
  labs(
    x=&quot;A Matrix&quot;,
    y=&quot;G Matrix&quot;,
    title= &quot;Off Diagonals&quot;
  ) +
  theme_minimal()</code></pre>
<p><img src="figure/pset2_tpm.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" />
Once difference that we see between the off-diagonal elements of the A
and G matrices are that the A matrix elements take on discrete values
where as the element from the G matrix are continuous. This is because
the A matrix calculates expectations of coefficients of coancestry based
on discrete assumptions of the starting founders having defined
relatedness. Those starting assumptions fix discrete expectation values
that any indiiduals can take at any level of relatedness. Since the GRM
is the realized additive relationship matrix, each value deviates from
the expectation, creating continuous values of additive relationship
coefficients.</p>
</div>
<div id="question-4" class="section level2">
<h2>Question 4</h2>
<p>Plot the diagonals of the pedigree relationship matrix (x-axis) and
the genomic relationship matrix (y-axis). Identify at least 1 difference
and provide an explanation (2 pts)</p>
<pre class="r"><code>plotting_data &lt;- data.frame(x=diag(a_mat), y=diag(G))
ggplot(data=plotting_data,aes(x=x, y=y))+
  geom_point(color=&quot;darkgreen&quot;) +
  labs(
    x=&quot;A Matrix&quot;,
    y=&quot;G Matrix&quot;,
    title= &quot;Diagonals&quot;
  ) +
  theme_minimal()</code></pre>
<p><img src="figure/pset2_tpm.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>One major difference that we see between the diagonals of the A and G
matrix diagonals is that in the A matrix, values cannot extend below 0
whereas in the GRM, values dip as low as about 0.69. This is primarily
due to the fact that the A matrix does not allow expected values of
coefficients of coancestry to extend below 0, wheras in the realized G
matrix, these can definitely fall below 0. This leads to negative values
of inbreeding coefficients which create diagonal elements in the GRM
that are below 1.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: aarch64-apple-darwin20
Running under: macOS 15.3.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] here_1.0.1      lubridate_1.9.4 forcats_1.0.0   stringr_1.5.1  
 [5] dplyr_1.1.4     purrr_1.0.2     readr_2.1.5     tidyr_1.3.1    
 [9] tibble_3.2.1    ggplot2_3.5.1   tidyverse_2.0.0

loaded via a namespace (and not attached):
 [1] sass_0.4.9        generics_0.1.3    stringi_1.8.4     hms_1.1.3        
 [5] digest_0.6.37     magrittr_2.0.3    evaluate_1.0.1    grid_4.4.1       
 [9] timechange_0.3.0  fastmap_1.2.0     rprojroot_2.0.4   workflowr_1.7.1  
[13] jsonlite_1.8.9    promises_1.3.2    scales_1.3.0      jquerylib_0.1.4  
[17] cli_3.6.3         crayon_1.5.3      rlang_1.1.4       bit64_4.5.2      
[21] munsell_0.5.1     withr_3.0.2       cachem_1.1.0      yaml_2.3.10      
[25] parallel_4.4.1    tools_4.4.1       tzdb_0.4.0        colorspace_2.1-1 
[29] httpuv_1.6.15     vctrs_0.6.5       R6_2.5.1          lifecycle_1.0.4  
[33] git2r_0.35.0      bit_4.5.0.1       fs_1.6.5          vroom_1.6.5      
[37] pkgconfig_2.0.3   pillar_1.10.0     bslib_0.8.0       later_1.4.1      
[41] gtable_0.3.6      glue_1.8.0        Rcpp_1.0.13-1     xfun_0.49        
[45] tidyselect_1.2.1  rstudioapi_0.17.1 knitr_1.49        farver_2.1.2     
[49] htmltools_0.5.8.1 labeling_0.4.3    rmarkdown_2.29    compiler_4.4.1   </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
