---
title: "Reproducible scripting, AlphaSimR, Homework"
author: 
  name: "Jean-Luc Jannink"
  affiliation: "USDA-ARS / Cornell"
date: "February 4, 2025"
output:
  html_document:
    fig_caption: yes
    fig_height: 6
    fig_width: 12
    highlight: haddock
    theme: cosmo
    toc: yes
    toc_float:
      collapsed: no
      toc_depth: 2
editor_options: 
  chunk_output_type: console
---
  
<style>
pre {
  overflow-x: auto;
}
pre code {
  word-wrap: normal;
  white-space: pre;
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo=TRUE)
knitr::opts_chunk$set(message=FALSE)
knitr::opts_chunk$set(cache=FALSE)
```

# Preliminaries

### Rmarkdown
This lab is written in Rmarkdown. I find it challenging to remember the syntax,
but here is a
[cheatsheet](https://rstudio.github.io/cheatsheets/html/rmarkdown.html)

### Learning objectives
1. A bit more on reproducible scripting  
Some other things that matter to me to ensure communicability and reproducibility
  i) The main README.md file should contain information about the conditions under which the script was run  
  ii) I like the script to have a predictable strucure: load and document the packages used, if there is a stochastic component, set the random seed, set and document the parameter values that will affect the script output  
  iii) Use a consistent directory structure. Happily, such a consistent structure is enforced by `workflowr`  
  iv) Use the package `here` to make file locations relative to the base project folder  

2. AlphaSimR  
`AlphaSimR` is a package to simulate breeding populations and tasks. It is not
completely intuitive nor completely well-documented. We will want to use it
ultimately to optimize breeding schemes. As a way to introduce AlphaSimR, this
Rmarkdown script contains code to illustrate the "outbreak of variation" that
occurs when a heterozygous individual is self-fertilized. This phenomenon was
important in supporting the notion that quantitative traits were influenced by
many loci.  Here is an 
[introduction](https://cran.r-project.org/web/packages/AlphaSimR/vignettes/intro.html)
showing code to simulate a very simple mass selection program.  Here is a
[deep dive](https://cran.r-project.org/web/packages/AlphaSimR/vignettes/traits.pdf)
into how `AlphaSimR` simulates traits.

3. Prepare a Homework  
Your homework will be to write a script that illustrates the interesting phenomenon of regression toward the mean between parents and progeny that we discussed in class.  

# Ordering of the script  
It's good to have all scripts in the same order with respect to standard tasks  

### Load packages first  
If your script depends on external packages, load them at the beginning. This shows users early on what the script dependencies are.  
```{r load packages, message=TRUE}
packages_used <- c("AlphaSimR", "tidyverse", "workflowr", "here")
ip <- installed.packages()
all_packages_installed <- TRUE
for (package in packages_used){
  if (!(package %in% ip[,"Package"])){
    print(paste("Please install package", package))
    all_packages_installed <- FALSE
  }
}#END packages_used
if (!all_packages_installed) stop("Need to install packages")

library(tidyverse)
```
Notice the conflicts report from loading tidyverse.  Two packages (`dplyr` and `stats`) both have a function called `filter`. Since dplyr was loaded *after* stats, if you use the function `filter`, it will go to the dplyr version.  It is dangerous to rely on what order packages have been loaded to determine which `filter` function you get.  R syntax to prevent ambiguity is to write either `dplyr::filter` or `stats::filter`.  Using that syntax will make your code more reproducible.  

### Set file locations relative to the project folder using `here`  
```{r set here path}
here::i_am("analysis/Lab3_ProgrammingAlphaSimR.Rmd")
```
There are two functions I use in the `here` package: `here::i_am` and `here::here`. `here::i_am` helps `here` "understand" the folder structure. You give it the name of the script that is currently running, and the path to that script from the base of the overall analysis folder. From that, `here` deduces where the base of the folder is.  After that, `here::here` locates files from that folder base. If you use the `here` package, you can send anyone a complicated zipped folder, and they should be able to run the analysis out of the box, regardless of where they put the folder in their own file system.  

### Document packages used  
This chunk creates a "README.md" file that documents the packages and versions used for future reference.  
```{r document packages}
source(here::here("code/addToREADME.R"))
addToREADME(paste0("## ", rmarkdown::metadata$title), append=F)
addToREADME(c(date(), ""))

packages_info <- ip[packages_used, c("Package", "Version", "Built")]
addToREADME(c("The packages used in this script are:", "Package, Version, Built"))
apply(packages_info, 1, function(vec) addToREADME(paste(vec, collapse=" ")))
addToREADME("")
```
Hyperlink the [README file](README.md) to the report.  That makes it easy to find.  

### Set random seed  
AlphaSimR generates many random numbers (e.g., to simulate Mendelian random segregation).  If you want the result of the analysis to come out the same each time (there are pros and cons) you need to set the random seed. Note that `workflowr` does this by default. If you are not using that package, then be explicit.  
```{r set random seed}
random_seed <- 45678
set.seed(random_seed)
addToREADME(c(paste("The random seed is", random_seed), ""))
```

### Script parameters  
If the behavior of your script depends on parameters that you set, initialize them early on.  
```{r simulation parameters}
nFounders <- 100
nChr <- 10 # Number of chromosomes
nSitesPerChr <- 1000 # Number of segregating sites _per chromosome_
nQTLperChr <- 10 # Vary this parameter to get oligo- versus poly- genic traits
nF1s <- 200 # We are going to make F1s to test outbreak of variation
nF2s <- 200 # We are going to make F2s to test outbreak of variation
```

### Parameters to README  
It's good to have all the information together in the README.  Chunks of code like this do not need to be included in the report.  To exclude them use the "include=FALSE" option in the chunk.  
```{r parameter to readme, include=FALSE}
addToREADME(c(
  paste("The number of founders for the simulation is", nFounders), 
  paste("The number of chromosomes is", nChr), 
  paste("There are", nQTLperChr, "QTL per chromosome"),
  paste("There are", nSitesPerChr, "segregating sites per chromosome"),
  paste("There are", nF1s, "F1 progeny"),
  paste("There are", nF2s, "F2 progeny"), ""
  ))
```

# Simulating some classical results  
This script uses `AlphaSimR` to simulate the "outbreak of variation" that arises when you self-fertilize a hybrid.  

### AlphaSimR populations  
The basic object of `AlphaSimR` is the population. To make founders, you first make founder haplotypes from a coalescent simulation, then you define simulation parameters that will link their genetic variation to phenotypic variation, then you make a first diploid population from the founder haplotypes.  
```{r Make founder population}
# Create haplotypes for founder population of outbred individuals
# Note: default effective population size for runMacs is 100
founderHaps <- AlphaSimR::runMacs(nInd=nFounders, nChr=nChr, 
                                  segSites=nSitesPerChr)
# founderHaps <- AlphaSimR::quickHaplo(nInd=nFounders, nChr=nChr, 
#                                  segSites=nSitesPerChr)

# New global simulation parameters from founder haplotypes
SP <- AlphaSimR::SimParam$new(founderHaps)
# Additive trait architecture
# By default, the genetic variance will be 1
SP$addTraitA(nQtlPerChr=nQTLperChr)

# Create a new population of founders
founders <- AlphaSimR::newPop(founderHaps, simParam=SP)
str(founders)
```

### Population information  
The population has `id`s. The `@mother` and `@father` ids are all zero because this population was made from founder haplotypes, and so does not have diploid parents. The genotypic values `gv` of the population are calculated for the trait created using `SP$addTraitA(nQtlPerChr=nQTLperChr)`. Given that there are `r nChr` chromosome`r ifelse(nChr > 1, "s", "")` and `r nQTLperChr` QTL per chromosome, there are `3^(nChr*nQTLperChr)` = `r 3^(nChr*nQTLperChr)` different possible genotypic values. The realized genotypic values are accessible with the function `gv(founders)`

From here, you can treat this population like a named vector using the square braces extraction operator `[ ]`.  Extract individuals by their `@id` or just by their order in the population using an integer index. For example, pick three random individuals from a population and list their ids. Pick the one with the first id in alphabetical order.  
```{r Population extraction operator}
test <- founders[c(2, 3, 5, 8, 13)]
testID <- test@id
firstID <- testID[order(testID)][1]
alphaInd <- test[firstID] # Extract individual with the first ID in order
print(testID)
print(alphaInd)
print(alphaInd@id)
```

### Outbreak of variation  
Emerson and East (1913) showed that if you crossed two inbreds, the hybrid had similar variation to each inbred, but if you then selfed the hybrid, the offspring varied more substantially. This code simulates that result. First, self the founders to homozygosity. The function `self` self-fertilizes individuals from the population. By default, it creates one selfed individual per parent (controllable with the parameter `nProgeny`), so this works nicely for single-seed descent.  
```{r self founders}
# Self-pollinate to for a few generations
nGenSelf <- 3
inbredPop <- founders
for (gen in 1:nGenSelf){
  inbredPop <- AlphaSimR::self(inbredPop)
}
```

### Check homozygosity  
Just a sanity check that this has, in fact, created a population of 100 individuals that are appropriately homozygous.  Loci are coded 0, 1, 2. So `qtl == 1` represents the case were a locus is heterozygous. `sum(qtl == 1)` counts those cases.  
```{r Check homozygosity}
qtl <- AlphaSimR::pullQtlGeno(inbredPop)
if (nrow(qtl) != nFounders) stop("The number of individuals is unexpected")
if (ncol(qtl) != nChr * nQTLperChr) stop("The number of QTL is unexpected")
fracHet <- sum(qtl == 1) / (nFounders * nChr * nQTLperChr)
cat("Expected fraction heterozygous", 1 / 2^nGenSelf, "\n",
    "Observed fraction heterozygous", fracHet, "+/-", 
    round(2*sqrt(fracHet*(1-fracHet)/(nFounders*nChr*nQTLperChr)), 3), "\n")
```

> What was wrong with my reasoning about the Expected fraction heterozygous?

Not all of the QTL loci started out as heterozygous in the founders. 

### Simulate outbreak of variation  
We will  
1. pick a random pair of inbred individuals  
2. cross that pair  
3. find out the variation in genotypic value among the pair's progeny  
4. pick a random F1 progeny  
5. self-fertilize that F1  
6. find out the variation in genotypic value among the F1's progeny  
*It's often useful to write very high-level outlines of what you want your code to do before you write the code*  
We will assume a trait that has a heritability of 0.5 in the base, non-inbred population. An easy way to get that heritability is with genetic and error variances of 1.  

### Pick random pair to make F1 hybrid  
```{r Pick random pair}
randomPair <- inbredPop[sample(nFounders, 2)]
```

### Cross the pair to make population of F1s  
The crossPlan is a **two-column matrix** with as many rows as the number of crosses you want to make, the first column is the `@id` or the index of the seed parent, and likewise for the pollen parent in the second column.  We will make `r nF1s` F1s from this random pair, so the matrix has `r nF1s` rows. You just want to cross individual 1 with individual 2, so each row contains 1:2.  
```{r Make hybrids}
crossPlan <- matrix(rep(1:2, nF1s), ncol=2, byrow=T)
f1_pop <- AlphaSimR::makeCross(randomPair, crossPlan)
head(crossPlan)
```

### Find out the phenotypic variation among F1s  
When you first make a population, `AlphaSimR` does not assume it has been phenotyped. You can phenotype it using the `setPheno` function.  Note that if you use `setPheno` on the same population a second time, that will overwrite the phenotypes from the first time. The genotypic variance can be retrieved using the function `varG`. Really, `varG` gives all we need to know, but of course that variance is generally not observable in non-simulated reality.  
```{r Phenotype F1s}
f1_pop <- AlphaSimR::setPheno(f1_pop, varE=1)
cat("Genotypic variance among F1s", round(AlphaSimR::varG(f1_pop), 3), "\n")
cat("Phenotypic variance among F1s", round(AlphaSimR::varP(f1_pop), 3), "\n")
hist(AlphaSimR::pheno(f1_pop), main="Histogram of F1 Phenotypes")
```

### Pick random F1  
```{r Pick random F1}
randomF1 <- f1_pop[sample(nFounders, 1)]
```

### Make F2 and observe phenotypic variance  
When you first make a population, `AlphaSimR` does not assume it has been phenotyped. You can phenotype it using the `setPheno` function.  Note that if you use `setPheno` on the same population a second time, that will overwrite the phenotypes from the first time. The genotypic variance can be retrieved using the function `varG`. Really, `varG` gives all we need to know, but of course that variance is generally not observable in non-simulated reality.  
```{r Make F2}
f2_pop <- AlphaSimR::self(randomF1, nProgeny=nF2s)
f2_pop <- AlphaSimR::setPheno(f2_pop, varE=1)
cat("Genotypic variance among F2s", round(AlphaSimR::varG(f2_pop), 3), "\n")
cat("Phenotypic variance among F2s", round(AlphaSimR::varP(f2_pop), 3), "\n")
hist(AlphaSimR::pheno(f2_pop), main="Histogram of F2 Phenotypes")
```

### Illustrate `here::here` by writing a file  
Just to practice again using the here package because it makes life easy.
```{r write QTL with here}
utils::write.csv(qtl, here::here("output", "QTLgenotypes.csv"))
```

# Homework  
Illustrate regression to the mean between parents and offspring using AlphaSimR  
1. You know how to create a population -- create a population of parents 

I am going to try using the inbred=TRUE argument in runMacs() to see the impact this has.

```{r Create Population of Parents}
set.seed(1234)

nFounders_b <- 200
nChr <- 10
nSitesPerChr <- 1000
parents_haps <- AlphaSimR::runMacs(nFounders_b, nChr, nSitesPerChr, inbred = FALSE)
SP_b <- AlphaSimR::SimParam$new(parents_haps)
SP_b$addTraitA(nQtlPerChr = 100)
parents <- AlphaSimR::newPop(parents_haps,SP_b)
```

2. You know how to get phenotypes from that population  

```{r}
par_pop <- AlphaSimR::setPheno(parents, varE=1,simParam = SP_b)
hist(par_pop@pheno, main="Parent Phenotypes", xlab = "Phenotype")
```

3. You want to randomly mate that population to get progeny that will be (we assume)
regressed to the mean. Create a `crossPlan` matrix like I did to generate the
F1s, except that each row should have randomly-picked parents, rather than 1 and
2 like for the F1s. There is also a command `AlphaSimR::randCross`.  Check out
its documentation. If you use that command, you will have to find the pedigree
of the progeny using the `@mother` and `@father` ids of the progeny population
and match those up to the parent population. If you make the crossPlan, then it
gives you the seed and pollen parent ids.

```{r}
nF1s_b <- 200
pollen_parents <- sample(min(par_pop@iid):max(par_pop@iid), nF1s_b, replace = TRUE)
seed_parents <- sample(min(par_pop@iid):max(par_pop@iid), nF1s_b, replace = TRUE)
crossPlan_b <- as.matrix(cbind(pollen_parents,seed_parents))

f1_pop_b <- AlphaSimR::makeCross(par_pop,crossPlan = crossPlan_b, simParam = SP_b)
```

4. Having made the progeny population, phenotype it also  

```{r}
f1_pop_b_pheno <- AlphaSimR::setPheno(f1_pop_b, varE=1, simParam = SP_b)
hist(f1_pop_b_pheno@pheno, main="F1 Phenotypes", xlab="Phenotypes")
```

5. Use each row of the `crossPlan` to find the two parents and calculate their phenotypic mean 

```{r}
crossPlan_b <- as.data.frame(crossPlan_b)
colnames(crossPlan_b)=c("pollen_parent","seed_parent")
crossPlan_b$seed_parent_pheno <- NA
crossPlan_b$pollen_parent_pheno <- NA
for (i in 1:nrow(crossPlan_b)){
  seed_parent <- par_pop[crossPlan_b[i,"seed_parent"]]
  pollen_parent <- par_pop[crossPlan_b[i,"pollen_parent"]]
  crossPlan_b[i,"seed_parent_pheno"] <- seed_parent@pheno
  crossPlan_b[i,"pollen_parent_pheno"] <- pollen_parent@pheno
}
crossPlan_b$midparent_value <- rowMeans(cbind(crossPlan_b$seed_parent_pheno,crossPlan_b$pollen_parent_pheno))

phenotypes <- as.data.frame(cbind(crossPlan_b$midparent_value, f1_pop_b_pheno@pheno))
colnames(phenotypes) <- c("midparent_value", "progeny_pheno")
```

6. Make a scatterplot of the progeny phenotypes against the parent mean phenotypes  

```{r}
ggplot(data=phenotypes, aes(x=midparent_value, y=progeny_pheno)) +
  geom_point()
```


> What is the regression coefficient?  

```{r}
lin.mod <- lm(progeny_pheno~midparent_value, data=phenotypes)
reg_coef <- lin.mod$coefficients[2]
print(reg_coef)
```

The regression coefficient of progeny phenotypes regressed on midparent values is 0.4784871. 

> How much closer to the mean are offspring, on average, compared to their parents?  

```{r}
mean_pheno <- mean(c(phenotypes$progeny_pheno,par_pop@pheno))
print(mean_pheno)
phenotypes$midpar_dev <- phenotypes$midparent_value-mean_pheno
avg_midpar_dev <- mean(phenotypes$midpar_dev)
phenotypes$prog_dev <- phenotypes$progeny_pheno-mean_pheno
avg_prog_dev <- mean(phenotypes$prog_dev)
avg_midpar_dev-avg_prog_dev
```

On average, progeny were 0.1455552 units closer to the average phenotype in the population than their parents. 

### Homework grading  
I want to receive a zipped folder with your homework in it one week from today, by midnight.  If you use the `workflowr` package to create the folder structure, that's great. You should have created an Rmarkdown script that will create an html file that I can look through. So:  
1. 4 points for just sending in a zipped folder on time  
2. 1 points for each successful step from above, with partial credit  
3. I will subtract 0.5 points for every day the homework is late  
